Userland/Libraries/LibGUI/UndoStack.h

```diff
@@ -35,6 +38,7 @@ class UndoStack {
 
     NonnullOwnPtrVector<UndoCommandsContainer> m_stack;
     size_t m_stack_index { 0 };
+    size_t m_clean_index { 0 };
```

Comment:
I think this needs to be `Optional<size_t>` because there will not always be a valid clean index.

Imagine the following sequence:
* Start (stack_index = 0, clean_index = {})
* Push command (stack_index = 1, clean_index = {})
* Push command (stack_index = 2, clean_index = {})
* Push command (stack_index = 3, clean_index = {})
* Mark stack clean (stack_index = 3, clean_index = 3)
* Undo one command (stack_index = 2, clean_index = 3)
* Undo one command (stack_index = 1, clean_index = 3)
* Push command (stack_index = 1, clean_index = ???)

At that point, I think the clean index should simply be invalid.

Also this is making me wonder if `m_stack_index` should also be `Optional<size_t>`. It's quite weird that it starts out with `0`, even though there are no entries in the stack!

Reply:
I guess we'll have to rewrite the entire undo stack then... I was avoiding doing that.
I'll test your scenario to see if it breaks the way it is.

Reply:
Yeah, we have a bug there 👍🏻

Reply:
I fixed the bug using `Optional` as you suggested but left the `m_stack_index` as it is for now, since it's not causing any problems and is out of the scope of this PR anyway.

(you taught me not to make drive-by fixes 😄)